archivebox/core/utils.py

```diff
@@ -1,39 +1,71 @@
-from pathlib import Path
-
 from django.utils.html import format_html
 
-from core.models import Snapshot
+from core.models import Snapshot, EXTRACTORS
 
 
 def get_icons(snapshot: Snapshot) -> str:
+    archive_results = snapshot.archiveresult_set
     link = snapshot.as_link()
     canon = link.canonical_outputs()
-    out_dir = Path(link.link_dir)
+    output = ""
+    output_template = '<a href="/{}/{}" class="exists-{}" title="{}">{} </a>'
+    icons = {
+        "singlefile": "❶",
+        "wget": "🆆",
+        "dom": "🅷",
+        "pdf": "📄",
+        "screenshot": "💻",
+        "media": "📼",
+        "git": "🅶",
+        "archive_org": "🏛",
+        "readability": "🆁",
+        "mercury": "🅼",
+    }
+    exclude = ["favicon"]
+    # Missing specific entry for WARC
+
+
+    for extractor in EXTRACTORS:
+        result = archive_results.filter(extractor=extractor[0])
```

Comment:
you should probably also filter out anything that isn't `status='succeeded'` (e.g. status=failed or status=skipped)